app/components/header.tsx

```diff
@@ -94,8 +99,25 @@ export default function Header() {
 
       <Flex justifyContent="flex-end" alignItems="center" color="white" gap="5" width="100%">
         <Hide below="lg">
-          <Text id="header-user">{user?.username}</Text>
+          <Text id="header-user">
+            {user.username === originalUser.username
+              ? user?.username
+              : `Mocking: ${user?.username}`}
```

Comment:
I'm not sure "Mocking" is the right term here.  We could use "Logged in As:" or "Impersonating:"